recipes-devtools/rust/files/rust-1.24.0/0001-librustc-always-allow-unstable-options.patch

```diff
@@ -0,0 +1,43 @@
+From 6a82f31d21ac7b85211e580585cc73ab2bdb0bc9 Mon Sep 17 00:00:00 2001
+From: Tyler Hall <tyler.hall@lexmark.com>
+Date: Sun, 29 Oct 2017 16:29:03 -0400
+Subject: [PATCH] librustc: always allow unstable options
```

Comment:
Can we have Upstream status here? And SOB as well.

Reply:
There is no upstream status here at all. As far as I know, this patch was never submitted upstream. The only reason I carried it forward from the Rust 1.21 build is that the Rocko has `-Zremap-path-prefix-from` added to the Rust flags for the patched that only works with the patched 1.21 compiler. I would be more than happy to rip that and the passing of the unstable flag out of the `rust-common.bbclass`, but I figured others would be unhappy about that.

Reply:
Well in that case we should use an upstream status as innapropiate as this sounds like a specific configuration we decided upon.

Reply:
You're only allowed to use "unstable" options on a nightly compiler because Rust guarantees that the compiler CLI will remain stable. The goal was to make reproducible builds which were needed by @srwalter and @tylerwhall. It was merged by them in #180. Effectively this means that you can use "unstable" options with the stable compiler generated by these recipes. This change will never be accepted upstream.

Reply:
I would say some of your review comments are issues with the #180 PR and you'd need to reach out to them to get a SOB. @jcreekmore isn't able to provide that. He's copying this patch into a named directory for Rust 1.24.0 and could instead refer to the old one instead of updating it.

Reply:
This explanation should reside in the patch. As well I propose to add an Inappropriate Upstream status as details in http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines .

Reply:
I don't disagree with you. None of the people involved right now had anything to do with that PR however.

Reply:
You could argue we could drop the patch until it meets guidelines. But I would say you'd want to update the project guidelines at https://github.com/meta-rust/meta-rust/wiki

Reply:
No, I understand now that these were carried on from an older version. Initially I was missing that bit. So given this situation I would say leave this as it is and hopefully somebody would address it in a follow up PR. Because otherwise we will be dragging patches without any understanding and somebody reading an update PR would be completely out of context.

Reply:
OK, I will leave it as it, then.